crates/atuin-client/src/database.rs

```diff
@@ -144,10 +144,21 @@ impl Sqlite {
             .with_regexp()
             .create_if_missing(true);
 
-        let pool = SqlitePoolOptions::new()
-            .acquire_timeout(Duration::from_secs_f64(timeout))
-            .connect_with(opts)
-            .await?;
+        let mut pool_opts = SqlitePoolOptions::new()
+            .acquire_timeout(Duration::from_secs_f64(timeout));
+
+        // Workaround for sqlx-sqlite in-memory failure.
+        // Make sure a single connection is created to avoid overwriting the database.
+        // Safe to use in this case since memory databases are only used for testing.
+        // https://github.com/launchbadge/sqlx/issues/2510
+        if path.to_str().unwrap().ends_with(":memory:"){
```

Comment:
I wanted to use [`opts.in_memory()`](https://github.com/launchbadge/sqlx/pull/3137) but that will only come in `sqlx 0.8